packages/cache/src/cache.ts

```diff
@@ -153,6 +201,104 @@ export async function restoreCache(
   return undefined
 }
 
+/**
+ * Restores cache using the new Cache Service
+ *
+ * @param paths a list of file paths to restore from the cache
+ * @param primaryKey an explicit key for restoring the cache
+ * @param restoreKeys an optional ordered list of keys to use for restoring the cache if no cache hit occurred for key
+ * @param downloadOptions cache download options
+ * @param enableCrossOsArchive an optional boolean enabled to restore on windows any cache created on any platform
+ * @returns string returns the key for the cache hit, otherwise returns undefined
+ */
+async function restoreCacheV2(
```

Comment:
for the `*CacheV2` functions, do we want to throw `GHESNotSupportedError`?

Reply:
I added a check to always default to v1 on GHES